group_vars/all

```diff
@@ -20,7 +19,26 @@ maas_snap_channel: "stable"     # if using snap, then the channel. ie: stable, b
 maas_deb_state: "present"       # if using deb, the state for install
 maas_state: "install"           # whether to install, or upgrade maas
 
-# grafana agent repo
+# O11y variables
+o11y_prometheus_url: ""
+o11y_loki_url: ""
+o11y_enabled_roles:
+  - maas_postgres_primary
+  - maas_postgres_secondary
+  - maas_region_controller
+  - maas_rack_controller
+
+o11y_enable: "{{ (o11y_prometheus_url|length > 0 or o11y_loki_url|length > 0) and (group_names|intersect(o11y_enabled_roles)|length>0) }}"
```

Comment:
This seems like enabling metrics is a little more implicit than most other optional things in the playbook. Perhaps have the URLs template in for specific versions and have `o11y_enable` be a boolean value that the user can set. (Makes it easy to add as a host variable too in case the user wants to enable on one host and not another).

Reply:
This is the way it was described in the spec, but I agree it looks better if we follow the pattern of other optional features. Updated the PR to reflect this